policies/api/src/main/java/org/eclipse/ditto/policies/api/commands/sudo/CheckPolicyPermissionsResponse.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.ditto.policies.api.commands.sudo;
+
+import static org.eclipse.ditto.base.model.common.ConditionChecker.checkNotNull;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import org.eclipse.ditto.base.model.common.HttpStatus;
+import org.eclipse.ditto.base.model.headers.DittoHeaders;
+import org.eclipse.ditto.base.model.json.JsonParsableCommandResponse;
+import org.eclipse.ditto.base.model.json.JsonSchemaVersion;
+import org.eclipse.ditto.base.model.signals.commands.AbstractCommandResponse;
+import org.eclipse.ditto.json.JsonFactory;
+import org.eclipse.ditto.json.JsonField;
+import org.eclipse.ditto.json.JsonFieldDefinition;
+import org.eclipse.ditto.json.JsonObject;
+import org.eclipse.ditto.json.JsonObjectBuilder;
+import org.eclipse.ditto.json.JsonPointer;
+import org.eclipse.ditto.json.JsonValue;
+import org.eclipse.ditto.policies.model.PolicyId;
+
+/**
+ * Response for a {@link CheckPolicyPermissions} command in the Ditto framework.
+ * <p>
+ * This class encapsulates the results of permission checks performed on various resources within a policy.
+ * The response contains a map of resource identifiers and corresponding boolean values indicating whether
+ * each permission was granted or denied.
+ * <p>
+ * This response is immutable and provides methods to build, parse from JSON, and convert permission results
+ * to and from a map format.
+ *
+ * @since 3.7.0
+ */
+@Immutable
+@JsonParsableCommandResponse(type = CheckPolicyPermissionsResponse.TYPE)
+public final class CheckPolicyPermissionsResponse extends AbstractCommandResponse<CheckPolicyPermissionsResponse>
+        implements PolicySudoQueryCommandResponse<CheckPolicyPermissionsResponse> {
+
+    /**
+     * The type of this response.
+     */
+    public static final String TYPE = "checkPolicyPermissionsResponse";
```

Comment:
Wrong "TYPE" string - responses must also use their `TYPE_PREFIX` + the command name, so here:
```
public static final String TYPE = TYPE_PREFIX + CheckPolicyPermissions.NAME;
```